File: backend/fastapi_face_engine/create_users.py
# create_users.py
from sqlalchemy.orm import Session
from .database import SessionLocal, engine
from . import crud, models, schemas

# Create tables using engine from database.py
models.Base.metadata.create_all(bind=engine)

# Users to create
users = [
    {"username": "admin", "password": "admin123", "category": "admin"},
    {"username": "teacher1", "password": "teacher123", "category": "teacher"},
    {"username": "student1", "password": "student123", "category": "student"},
    {"username": "finance1", "password": "finance123", "category": "finance"}
]

db: Session = SessionLocal()

for u in users:
    existing_user = db.query(models.User).filter(models.User.username == u["username"]).first()
    if not existing_user:
        crud.create_user(db, schemas.UserCreate(**u))

db.close()
print("Users created successfully!")

File: backend/createUsers.js
const mongoose = require("mongoose");
const bcrypt = require("bcryptjs");
const User = require("./models/User");
require("dotenv").config();

// Connect to MongoDB
mongoose.connect(process.env.MONGO_URI)
  .then(() => console.log("MongoDB connected"))
  .catch(err => console.error(err));

const users = [
  { username: "admin1", password: "admin123", role: "admin" },
  { username: "teacher1", password: "teacher123", role: "teacher" },
  { username: "student1", password: "student123", role: "student" },
];

const createUsers = async () => {
  try {
    for (let u of users) {
      const hashedPassword = await bcrypt.hash(u.password, 10);
      const user = new User({ username: u.username, password: hashedPassword, role: u.role });
      await user.save();
      console.log(`User ${u.username} created`);
    }
    console.log("All users created successfully!");
  } catch (err) {
    console.error(err);
  } finally {
    mongoose.disconnect();
  }
};

createUsers();
